packages/inter-protocol/src/clientSupport.js

```diff
@@ -204,33 +204,28 @@ const makePsmSwapOffer = (instance, brands, opts) => {
 };
 
 /**
- * @param {Record<string, Brand>} brands
+ * @param {(amt: string) => Amount<'nat'>} parseAmount
  * @param {{
  *   offerId: string,
- *   collateralBrandKey: string,
- *   give: number,
- *   desiredBuy: number,
- *   wantMinimum?: number,
+ *   give: string,
+ *   desiredBuy: string,
+ *   wantMinimum?: string,
  * } & ({
  *   price: number,
  * } | {
  *   discount: number,  // -1 to 1. e.g. 0.10 for 10% discount, -0.05 for 5% markup
  * })} opts
  * @returns {import('@agoric/smart-wallet/src/offers.js').OfferSpec}
  */
-const makeBidOffer = (brands, opts) => {
-  /** @type {Brand<'nat'>} */
-  // @ts-expect-error XXX how to narrow AssetKind?
-  const collateralBrand = brands[opts.collateralBrandKey];
-  const makeCollateral = x =>
-    AmountMath.make(collateralBrand, scaleDecimals(x));
-
+const makeBidOffer = (parseAmount, opts) => {
```

Comment:
this makes this offer maker different than all the rest in here. I think we should maintain an ClientOfferMaker function signature. Can't enforce that until we upgrade to TS 5.0 though: https://devblogs.microsoft.com/typescript/announcing-typescript-5-0/#satisfies-support-in-jsdoc

I won't block on this but please leave an XXX comment or make a new issue to get consistency.

Reply:
added XXX

Reply:
on second thought, in order to address some testing feedback from CH above, I moved `parseAmount` in to `opts` so that this one follows the convention.